### PR TITLE
Generate the dcos superuser password hash via python in a docker container

### PR DESCRIPTION
**Pull request by tlake**
# Important note
Since these changes generate the hashed password using `docker run`, **`terraform apply` now prerequires that the user has setup Docker in the applying shell!**

```
$  docker-machine start NAME
$  eval $(docker-machine env NAME)
```
or something equivalent.

I'm not sure where we ought to document this, but it should be somewhere other than this comment. :p I'm happy to make that change where appropriate.

# Summary

These changes make use of Terraform's [external data sources](https://www.terraform.io/docs/providers/external/data_source.html) in order to obtain a value as the result of executing a shell script. This data source is defined in `aws/main.tf` and used twice in `aws/bootstrap.tf`. The script which is executed is `aws/scripts/local/hash-with-docker.sh`.

### CODE REVIEW

File: aws/main.tf [hunk 1]
```diff
@@ -7,14 +7,23 @@ provider "aws" {
...
 
 # Runs a local script to return the current user in bash
 data "external" "whoami" {
   program = ["scripts/local/whoami.sh"]
 }
 
+# Runs a local script to hash the superuser's password using a python library via `docker run`.
+# Reference the hashed value with "${data.external.dcos_superuser_password_hash.result.hash}"
+data "external" "dcos_superuser_password_hash" {
+    program = [ "scripts/local/hash-with-docker.sh" ]
```
[the remark this answers is not in the document]
Comment (tlake) [reply aws/main.tf]: Yeah! ✅

**Comment by nehayward** [comment]
We could just do a `which docker` to see if it exists.

**Comment by tlake** [comment]
@sesh-kebab @nehayward I'm not too familiar with docker implementations that aren't `docker-machine`, but I think the issue is maybe slightly bigger than "is Docker installed?". I needed to make sure that docker environment variables had been exported into the shell which was running the `terraform apply`. Is the workflow with, say, Docker for Mac particularly different?

**Comment by sesh-kebab** [comment]
> Is the workflow with, say, Docker for Mac particularly different
Yes. Once docker for mac is installed, i've never really had to think about docker env vars.

**Pull request merged**